### ISSUE TITLE
Add home to source paths

### ISSUE THREAD

**Issue by LiterallyUniqueLogin** [koalaman/shellcheck#1845]
#### For bugs
- Rule Id (if any, e.g. SC1090): 
- My shellcheck version (`shellcheck --version` or "online"): 0.7.0 
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC1090)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
I've set the flags 
```sh 
-P /home/username/ -P /
```

```sh

#!/bin/bash
source ~/.bashrc
```

#### Here's what shellcheck currently says:

SC1090: Can't follow non-constant source. Use a directive to specify location.

#### Here's what I wanted or expected to see:

I wanted shellcheck to find .bashrc in /home/username/.bashrc and not show a warning.
If I insert
```sh
 # shellcheck source=/home/username/.bashrc
```
above this line, shellcheck finds the file and doesn't show a warning (but if I misspell the file name then shellcheck still warns), so my overall usage of shellcheck isn't borked.

Is there any way I can use source paths or some other feature to tell shellcheck where my home will be without having to specify a directive each time?

**Comment by b0o** [comment]
I would appreciate this as well. There are often times where I need to source a file in the user's home directory, e.g. `~/.config/foo`, and I end up needing to hard-code my own home directory path to satisfy shellcheck. I feel this looks sloppy when shared with others (see my [neuron-autoindex](https://github.com/b0o/neuron-extras/blob/df83ac6dabe8446e50a86ddb9792bc5749bba705/neuron-autoindex#L8) script for an example).

**Comment by tycho-kirchner** [comment]
Just wanted to document here, that using `$HOME` instead of `~` works nicely (together with `-x -P $HOME`). 
Script test.sh:
~~~
#!/usr/bin/env bash
source "$HOME/.lib/bash_commons.sh" || exit
echo "scriptdir $scriptdir"  # scriptdir is set in my bash_commons.sh
~~~
Run shellcheck with:
~~~
shellcheck -x -P $HOME ./test.sh
~~~
s. also [here](https://github.com/koalaman/shellcheck/issues/769)

**Comment by ELLIOTTCABLE** [comment]
There's no way to set this in a `.shellcheckrc`, is there, since `$HOME` is being expanded by the shell in the above approach?

Unfortunately, that workaround doesn't work very well with IDEs … hrm
